akka-docs/src/main/paradox/additional/native-image.md

```diff
@@ -0,0 +1,86 @@
+ * Custom Mailbox Types
+ * Akka Persistence plugins
+ * Akka Discovery implementations
+ * Akka Lease implementations
+
+For plugins provided in libraries, metadata is preferably provided with the library so that end user applications
+does not need to provide metadata for them.
+
+### Serialization with Jackson
+
+When using the built-in @apidoc[JsonSerializable] and @apidoc[CborSerializable] marker traits, message types are automatically added
+for reflective access by Akka. 
+
+If instead using marker self-defined traits, the marker trait defined in `serialization-bindings` as well as each 
+concrete message types (lookup and constructor) needs to be added to the reflection metadata.
+
+Applications that define and configure @apidoc[JacksonMigration] to evolve data formats needs to list each concrete
+migration implementation (lookup and constructor) in reflection metadata.
+
+Additional object mappers added through config `akka.serialization.jackson.jackson-modules` needs entries in the reflective 
+metadata (lookup and constructor).
+
+### Third party serializers
+
+Third party serializers will need reflection metadata for the serializer implementation, the types used in `serialization-bindings`
+and possibly further per-message metadata depending on the specific serializer logic.
+
+### Extensions
+
+Classic and typed `Extension`s loaded via configuration (`akka.extensions`, `akka.actor.typed.extensions`, `akka.actor.library-extensions` or `akka.actor.typed.library-extensions`)
+needs an entry in the reflection metadata (lookup and constructor).
+
+### Akka Persistence Event Adapters
+
+Event adapters defined in an application needs to be listed in reflection metadata (lookup and constructor).
+
+### Reflective classic actor construction
+
+Classic actors that has a @apidoc[akka.actor.Props$] defined using the @scala[type `Props[T]()` or class `Props(classOf[T], ...)`] 
+@java[class `Props.create(T.getClass, ...)`] needs reflection entries for lookup and the constructor matching the set of passed parameters. 
+
+An easier path is to instead use lambda factories @scala[type `Props(new T)`]
+@java[class `Props.create(T.getClass, () -> new T())`] to define props. The only reason to use the reflection based approach
+is the classic remote deploy feature.
+
+### Logging
+
+When using `akka-slf4j` for logging, automatically used for `akka-actor-typed`, it is very likely that the concrete logger
+chosen needs extra configuration.
+
+While Akka does not mandate a logger implementation, `logback-classic` is used in many Akka samples throughout the Akka projects. 
+To make logback requires additional entries reflect entries, configuring static initalization of logback and either avoid using 
+`ch.qos.logback.classic.AsyncAppender` or declare your own lazy version of (not starting any threads at compile time).
+
+FIXME should we add link to working sample/config for logback?
+FIXME should we provide a lazy async logger appender or show it here?
```

Comment:
Both sound useful.

Reply:
I'll follow up with these two after this massive PR is merged.

Reply:
Another follow up will be to link to the other umbrella project doc pages to make it easy to find everything you need to know when building something using more Akka than just core.